File: cpp/leap/leap.cpp
#include "leap.h"

namespace leap {

bool is_divisible( int value, int divisor ) {

    return value % divisor == 0;

}

bool is_leap_year( int year ) {

    if( not is_divisible( year, 4 ) )
        return false;

    if( is_divisible( year, 400 ) )
        return true;

    if( is_divisible( year, 100 ) )
        return false;

    return true;

}

}  // namespace leap

File: java/leap/src/main/java/Leap.java
class Leap {

    private boolean isDivisible(int number, int divisor) {
        return number % divisor == 0;
    }

    boolean isLeapYear(int year) {
        return isDivisible(year, 4) && (!isDivisible(year, 100) || isDivisible(year, 400));
    }

}
